frontend/src/app/home/services/main.service.ts

```diff
@@ -28,31 +30,38 @@ export class MainService {
   room = [];
   kitchen = [];
   guest = [];
+  employee = [];
   all = [];
+  options = [];
+
 
   select$ = this.select.asObservable();
 
   constructor(private http: HttpClient, deskService: DeskService,
     officeService: OfficeService, guestService: GuestService,
     kitchenService: KitchenService, printerService: PrinterService,
-    roomService: RoomService) {
+    roomService: RoomService, employeeService: EmployeeService) {
 
     forkJoin(
       deskService.getDesk(),
       officeService.getOffice(),
       kitchenService.getKitchen(),
       printerService.getPrinter(),
       roomService.getRoom(),
-      guestService.getGuest()
-    ).subscribe(([desks, offices, kitchens, printers, rooms, guests]) => {
+      guestService.getGuest(),
+      employeeService.getEmployee(),
+    ).subscribe(([desks, offices, kitchens, printers, rooms, guests, employees]) => {
       this.desk = desks.map(x => new Desk(x));
       this.office = offices.map(x => new Office(x));
       this.kitchen = kitchens.map(x => new Kitchen(x));
       this.printer = printers.map(x => new Printer(x));
       this.room = rooms.map(x => new Room(x));
+      this.employee = employees.map(x => new Employee(x));
       this.guest = guests;
-
-      this.all = [...this.desk, ...this.office, ...this.kitchen, ...this.printer, ...this.room];
+      this.options = [...this.kitchen.map(x => x.name.toUpperCase()), ...this.room.map(x => x.roomName.toUpperCase()),
+      ...this.office.map(x => `${x.firstName.toUpperCase()} ${x.lastName.toUpperCase()}`),
+      ...this.employee.map(x => `${x.firstName.toUpperCase()} ${x.lastName.toUpperCase()}`)];
+      this.all = [...this.kitchen, ...this.room, ...this.office, ...this.desk, ...this.printer];
```

Comment:
Jeżeli chodzi o `options` to można by zrobić bardzo podobnie jak w modalu na dole tzn. modele miałyby funkcję analogiczną do `map` tylko zwracały by string po którym można by było szukać.